extensions/elation/actions/createNonVisitNote.ts

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ZodError } from 'zod'
+import {
+  FieldType,
+  type Action,
+  type DataPointDefinition,
+  type Field,
+} from '../../../lib/types'
+import { Category } from '../../../lib/types/marketplace'
+import { type settings } from '../settings'
+import { makeAPIClient } from '../client'
+import { fromZodError } from 'zod-validation-error'
+import { AxiosError } from 'axios'
+import { nonVisitNoteSchema } from '../validation/nonVisitNote.zod'
+
+const fields = {
+  text: {
+    id: 'text',
+    label: 'Text',
+    description: 'Text of a note',
+    type: FieldType.STRING,
+    required: true,
+  },
+  author: {
+    id: 'author',
+    label: 'Author',
+    description: 'Author of a note. Should be ID of a User.',
+    type: FieldType.NUMERIC,
+    required: true,
+  },
+  patient: {
```

Comment:
for consistency, should we call this `patientId`?

also, i notice it's numeric at times and a string in other fields.

e.g. getPatient:
![image](https://github.com/awell-health/awell-extensions/assets/19916684/206b6493-3e69-4696-9a98-c56e35f668dd)

getAppointment:
![image](https://github.com/awell-health/awell-extensions/assets/19916684/7262f3de-fdb0-4cf7-a1fa-5436d79ee138)

Reply:
^^ ignore this one for now, i'll put it in a separate issue